File: lib/src/main/java/fr/o80/lib/abtesting/ABStore.java
package fr.o80.lib.abtesting;

import android.content.SharedPreferences;

/**
 * @author Olivier Perez
 */
class ABStore {

    private SharedPreferences sharedPref;

    ABStore(SharedPreferences sharedPref) {
        this.sharedPref = sharedPref;
    }

    int getInt(String name, int defaultValue) {
        return sharedPref.getInt(name, defaultValue);
    }

    void set(String name, int value) {
        sharedPref.edit()
                .putInt(name, value)
                .apply();
    }
}

File: abtesting/src/main/java/fr/o80/lib/abtesting/ABStore.kt
package fr.o80.lib.abtesting

import android.content.SharedPreferences

/**
 * @author Olivier Perez
 */
class ABStore(private val sharedPref: SharedPreferences) {

    fun getInt(name: String, defaultValue: Int): Int =
            sharedPref.getInt(name, defaultValue)

    operator fun set(name: String, value: Int) {
        sharedPref.edit()
                .putInt(name, value)
                .apply()
    }
}
